src/wattsi.pas

```diff
@@ -1817,14 +1932,60 @@ procedure Save(const Document: TDocument; const FileName: AnsiString; const InSp
    var
       IsExcluder: Boolean;
       AttributeCount: Cardinal;
+      HighlighterOutput: TStringList;
    begin
       if (InSplit and Element.HasAttribute(kExcludingAttribute[vSplit])) then
          exit;
       IsExcluder := DetermineIsExcluder(Element, AttributeCount);
       if (not (IsExcluder or Element.HasProperties(propVoidElement) or
                (Element.HasProperties(propOptionalEndTag) and ((not Assigned(Element.NextSibling)) or (AutoclosedBy(Element, Element.NextSibling)))))) then
       begin
-         Write(F, '</', Element.LocalName.AsString, '>');
+         WriteJSON(']');
+         WriteHTML('</' + Element.LocalName.AsString + '>');
+         if (Element.IsIdentity(nsHTML, ePre) and (HighlighterExecutablePath <> '')) then
+         begin
+            Highlighter.Parameters.Clear;
+            // usage: __init__.py [-h] [--output {json,html}] [--numbers]
+            // [--highlights LINEHIGHLIGHTS] [--start LINESTART] [--unescape]
+            // [--just {html,css}] lang
+            Highlighter.Parameters.Add('--output');
+            Highlighter.Parameters.Add('html');
+            Highlighter.Parameters.Add('--unescape');
+            Highlighter.Parameters.Add('--just');
+            Highlighter.Parameters.Add('html');
+            if (Pos(Element.GetAttribute('class').AsString, 'idl') <> 0) then
+               Highlighter.Parameters.Add('webidl')
+            else
+            if (Pos(Element.GetAttribute('class').AsString, 'css') <> 0) then
```

Comment:
Should we include js in here as well for when we fix the source document?

Reply:
+            if (Pos(Element.GetAttribute('class').AsString, 'css') <> 0) then

> Should we include js in here as well for when we fix the source document?

Yes, added in 00969db